src/WebformHelper.php

```diff
@@ -0,0 +1,274 @@
+            '#type' => 'link',
+            '#title' => str_replace(urlencode($uuidPlaceholder), $uuidPlaceholder, $url->toString()),
+            '#url' => $url,
+            '#prefix' => '<li>',
+            '#suffix' => '</li>',
+          ];
+        }
+      }
+    }
+  }
+
+  /**
+   * Get webform by id or submission uuid.
+   *
+   * If submission uuid is specified (i.e. not null), the submission's webform's
+   * id must match the specified webform id.
+   *
+   * @return \Drupal\webform\WebformInterface|null
+   *   The webform if found.
+   */
+  public function getWebform(string $webformId, string $submissionUuid = NULL): ?WebformInterface {
+    if (NULL !== $submissionUuid) {
+      $storage = $this->entityTypeManager->getStorage('webform_submission');
+      $submissionIds = $storage
+        ->getQuery()
+        ->accessCheck(FALSE)
+        ->condition('uuid', $submissionUuid)
+        ->execute();
+      $submission = $storage->load(array_key_first($submissionIds));
+
+      if (NULL === $submission) {
+        return NULL;
+      }
+
+      assert($submission instanceof WebformSubmissionInterface);
+      $webform = $submission->getWebform();
+      if ($webformId !== $webform->id()) {
+        return NULL;
+      }
+
+      return $webform;
+    }
+
+    return $this->entityTypeManager
+      ->getStorage('webform')
+      ->load($webformId);
+  }
+
+  /**
+   * Get users allowed to access a webform's data.
+   *
+   * @return \Drupal\user\UserInterface[]|array
+   *   The users.
+   */
+  public function getAllowedUsers(WebformInterface $webform): array {
+    $settings = $webform->getThirdPartySetting('os2forms', 'os2forms_rest_api');
+    $allowedUserIds = $settings['allowed_users'] ?? [];
+
+    return $this->loadUsers($allowedUserIds);
+  }
+
+  /**
+   * Load users.
+   *
+   * @phpstan-param array<int, mixed> $spec
+   * @phpstan-return array<int, mixed>
+   */
+  private function loadUsers(array $spec): array {
+    return $this->entityTypeManager
+      ->getStorage('user')
+      ->loadMultiple(array_column($spec, 'target_id'));
+  }
+
+  /**
+   * Implements hook_file_download().
+   *
+   * @phpstan-return int|array<string, string>|null
+   */
+  public function fileDownload(string $uri) {
+    $request = \Drupal::request();
+
+    if ($user = $this->keyAuth->authenticate($request)) {
+      // Find webform id from uri, see example uri.
+      // @Example: private://webform/some_webform_id/119/some_file_name.png
+      $pattern = '/private:\/\/webform\/(?<webform>[^\/]*)/';
+      if (!preg_match($pattern, $uri, $matches)) {
+        // Something is not right, deny access.
+        return -1;
+      }
+
+      // User has API access.
+      $webform = \Drupal::entityTypeManager()->getStorage('webform')->load($matches['webform']);
+      $settings = $webform->getThirdPartySetting('os2forms', 'os2forms_rest_api');
+
+      $allowedUsers = $this->loadUsers($settings['allowed_users'] ?? []);
```

Comment:
can it be replaced with function `public function getAllowedUsers`?

Reply:
It can and should be. Fixed in https://github.com/OS2Forms/os2forms_rest_api/pull/1/commits/902d619d66d35033a4d67520ad40deee819b449a.